Scripts/Runtime/Entities/Tasks/AbstractTaskDriverSystem.cs

```diff
@@ -7,24 +7,47 @@
 
 namespace Anvil.Unity.DOTS.Entities
 {
+    public abstract partial class AbstractTaskDriverSystem<TKey, TTaskData> : AbstractTaskDriverSystem
+        where TKey : unmanaged, IEquatable<TKey>
+        where TTaskData : unmanaged, IKeyedData<TKey>, ITaskData
+    {
+        public new VirtualData<TKey, TTaskData> TaskData
+        {
+            get => base.TaskData as VirtualData<TKey, TTaskData>;
+        }
+
+        protected AbstractTaskDriverSystem()
+        {
+            base.TaskData = CreateTaskData<TKey, TTaskData>();
```

Comment:
probably better to pass this down through a constructor param.

Reply:
You're right but I can't easily do it right now due to not being able to call the `CreateTaskData` function in the constructor. 

I need to refactor this a bit, so I've added the TODO and linked back to here.